src/styling/CSSTemplate.ts

```diff
@@ -1,3 +1,5 @@
+import "../globalTypes/CSSStyleSheet.ts"
```

Comment:
Shouldn't this be `import type`?

Reply:
Apparently `import type` is not supported without importing something specific. However, the way it is in its current form is not ideal, because it'll be importing this file even at runtime if it's not a type import.

Reply:
This is more a way around using a `globalThis.d.ts` - it's just to include that type in the global scope

Saying this, where does it mention `replace` exists on `CSSStyleSheet`? Not the one ou define, but I remember you saying that the lib doesn't have it defined (yet) but it does exist, or something along those lines?

Reply:
> Saying this, where does it mention `replace` exists on `CSSStyleSheet`? Not the one ou define, but I remember you saying that the lib doesn't have it defined (yet) but it does exist, or something along those lines?

Here's [the MDN page](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet) and here's [some additional documentation](https://developers.google.com/web/updates/2019/02/constructable-stylesheets).

Reply:
The problem with `import "./path"` is that it shows up in the final compiled output, which is why type-only imports need to be imported with `import type` to avoid there being extraneous junk in the output.

Reply:
> > Saying this, where does it mention `replace` exists on `CSSStyleSheet`? Not the one ou define, but I remember you saying that the lib doesn't have it defined (yet) but it does exist, or something along those lines?
> 
> Here's [the MDN page](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet) and here's [some additional documentation](https://developers.google.com/web/updates/2019/02/constructable-stylesheets).

ahh ok i couldn't see an example for it (but i just didnt check the compat section), though it's not available on safari or firefox :/ (not sure if you're aware of that?)

Curious why it isn't included in the lib

Reply:
TS typically doesn't include types for "experimental" features until they have wide browser support. Users are not expected to use this API with the lib directly; the library uses it internally if it's supported. If it isn't, it falls back to using an older API that performs considerably worse but works in every browser.

Reply:
Ok gotcha, looked through the code and can understand this better

I don't think importing it as a type is a possibility, and if you didn't wanna avoid junk then i *think* the next best thing would be slapping a ts ignore on it :/ But ill keep thinking

Reply:
Or another 'hack' would be modifying that global type file to just export an interface, and do something like:

```ts
// globaltypes/cssstylesheet.ts
export intrface ICSSStyleSheet extends CSSStyleSheet {
  replace: (css: string) => void
}

// styling/CSSTemplate.ts
import type { ICSSStyleSheet } from ...

...

  (this.#stylehsheet as ICSSStyleSheet).replace(this.cssText)
```

Reply:
or do `#stylesheet: ICSSStyleSheet | undefined` instead (keeping the interface we import too)

Reply:
Seems like doing the cast might be the least bad option.

Reply:
https://github.com/0kku/destiny/pull/17/commits/2f82977d7b2da4228d437a2b12fe69841fc13164